### DISCUSSION TITLE
GitHub actions has such great statuses, what about pull request status checks?

### DISCUSSION THREAD

**Discussion by idan-cyber** [community/community#65366] [Pull Requests] [answered]
### Select Topic Area

Product Feedback

### Body

GitHub actions are indeed the coolest thing ever and as such the status of the workflow is very detailed (https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository):
Status can be one of: completed, action_required, cancelled, failure, neutral, skipped, stale, success, timed_out, in_progress, queued, requested, waiting, pending.

How about we take this greatness into the commit status checks in a pull request? Currently a check has only these statuses (https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#create-a-commit-status):
State can be one of: error, failure, pending, success.

In particular, splitting `pending` to `pending` and `in progress` would be super useful.

Thanks!

**Comment by fmigneault** [comment]
Having `skipped` as well would be great. 
Currently, one must either use `failed` or `success`, both of which indicate incorrect information that can be misleading.
When skipped, the operations are neither failed nor succeeded, they are simply not evaluated.

**Comment by idan-cyber** [comment, marked as answer]
GitHub referred me to their commit check runs APIs - https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28. Much more versatile and detailed than status checks.

**Reply by fmigneault** [reply]
In my case, a Jenkins plugin is triggered when a PR is opened or commits are added.
When triggered, a GitHub statuses entry with "pending" is automatically generated. 
If I manually POST to the `check-runs` endpoint instead of `statuses`, wouldn't that simply create duplicate "checks" in my PR?
